internal/host/host.go

```diff
@@ -320,23 +339,32 @@ func (m *Manager) RefreshStatus(ctx context.Context, h *models.Host, db *gorm.DB
 	if err != nil {
 		return err
 	}
-	conditions, validationsResults, err := m.rp.preprocess(vc)
+	conditions, newValidationRes, err := m.rp.preprocess(vc)
 	if err != nil {
 		return err
 	}
-	if err = m.reportValidationStatusChanged(ctx, vc, h, validationsResults); err != nil {
+	currentValidationRes, validationsChanged, err := m.checkValidationChanged(ctx, h, newValidationRes)
+	if err != nil {
 		return err
 	}
+	if validationsChanged {
+		m.reportValidationStatusChanged(ctx, vc, h, newValidationRes, currentValidationRes)
+	}
 	err = m.sm.Run(TransitionTypeRefresh, newStateHost(h), &TransitionArgsRefreshHost{
 		ctx:               ctx,
 		db:                db,
 		eventHandler:      m.eventsHandler,
 		conditions:        conditions,
-		validationResults: validationsResults,
+		validationResults: newValidationRes,
 	})
 	if err != nil {
 		return common.NewApiError(http.StatusConflict, err)
 	}
+	if validationsChanged {
```

Comment:
I think that we should add a comment why we are doing it after the transition

Reply:
Updated.